input/system/heroku/logs.go

```diff
@@ -215,27 +214,22 @@ func setupLogTransformer(ctx context.Context, wg *sync.WaitGroup, servers []*sta
 
 				now := time.Now()
 
-				var newLogLine *state.LogLine
+				var logLine *state.LogLine
 				var sourceName string
-				sourceToServer, newLogLine, sourceName = logStreamItemToLogLine(item, servers, sourceToServer, now, globalCollectionOpts, logger)
-				if newLogLine != nil && sourceName != "" {
-					logLinesBySource[sourceName] = append(logLinesBySource[sourceName], *newLogLine)
+				sourceToServer, logLine, sourceName = logStreamItemToLogLine(item, servers, sourceToServer, now, globalCollectionOpts, logger)
+				if logLine == nil || sourceName == "" {
+					continue
 				}
 
-				for sourceName, logLines := range logLinesBySource {
-					server, exists := sourceToServer[sourceName]
-					if !exists {
-						logger.PrintInfo("Ignoring log line since server can't be matched yet - if this keeps showing up you have a configuration error for %s", sourceName)
-						logLinesBySource[sourceName] = []state.LogLine{}
-						continue
-					}
-
-					for _, logLine := range logLines {
-						logLine.Username = server.Config.GetDbUsername()
-						logLine.Database = server.Config.GetDbName()
-						out <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
-					}
+				server, exists := sourceToServer[sourceName]
+				if !exists {
+					logger.PrintInfo("Ignoring log line since server can't be matched yet - if this keeps showing up you have a configuration error for %s", sourceName)
```

Comment:
Not really something we should fix here (especially since I don't have any better suggestions), but this is a pretty bad error message: "you have a configuration" error doesn't really tell you anything, since there's very little configuration for a Heroku-based collector.